#
# Add a missing Exception class to /scapy/arch/linux.py
# 
# To apply this patch:
# STEP 1: Chdir to the source directory. If you have scapy installed, this is
#         likely located at /usr/share/pyshared.
# STEP 2: Run the 'patch' program with this file as input, i.e.:
#
#    /usr/share/pyshared$ patch -p1 ./scapy/arch/linux.py </path/to/this/file>
#
# To apply this patch with the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# @authors: Isis Lovecruft
# @license: This file is part of ooniprobe, see LICENSE file for details.
# @copyright: 2012 Isis Lovecruft
#
#### End of Preamble ####

#### Patch data follows ####
diff -c 'scapy/arch/linux.py' 'scapy/arch/linux.py-patched'
Index: ./scapy/arch/linux.py
*** ./scapy/arch/linux.py	Thu Dec 13 21:26:59 2012
--- ./scapy/arch/linux.py-patched	Thu Dec 13 21:27:53 2012
***************
*** 14,21 ****
  from scapy.data import *
  from scapy.supersocket import SuperSocket
  import scapy.arch
! from scapy.error import warning
! 
  
  
  # From bits/ioctls.h
--- 14,20 ----
  from scapy.data import *
  from scapy.supersocket import SuperSocket
  import scapy.arch
! from scapy.error import warning,Scapy_Exception
  
  
  # From bits/ioctls.h
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Thu Dec 13 21:33:12 2012
# Generated by        : makepatch 2.03
# Recurse directories : Yes
# Excluded files      : (\A|/).*\~\Z
#                       (\A|/).*\.a\Z
#                       (\A|/).*\.bak\Z
#                       (\A|/).*\.BAK\Z
#                       (\A|/).*\.elc\Z
#                       (\A|/).*\.exe\Z
#                       (\A|/).*\.gz\Z
#                       (\A|/).*\.ln\Z
#                       (\A|/).*\.o\Z
#                       (\A|/).*\.obj\Z
#                       (\A|/).*\.olb\Z
#                       (\A|/).*\.old\Z
#                       (\A|/).*\.orig\Z
#                       (\A|/).*\.rej\Z
#                       (\A|/).*\.so\Z
#                       (\A|/).*\.Z\Z
#                       (\A|/)\.del\-.*\Z
#                       (\A|/)\.make\.state\Z
#                       (\A|/)\.nse_depinfo\Z
#                       (\A|/)core\Z
#                       (\A|/)tags\Z
#                       (\A|/)TAGS\Z
# p 'scapy/arch/linux.py' 16563 1355434073 0100644
#### End of ApplyPatch data ####

#### End of Patch kit [created: Thu Dec 13 21:33:12 2012] ####
#### Patch checksum: 56 1830 50583 ####
#### Checksum: 74 2507 41533 ####
